### DISCUSSION TITLE
[Question] How to read messages that was already received using `receive` command ?

### DISCUSSION THREAD

**Discussion by redradist** [AsamK/signal-cli#1284] [General]
Hi all,

I've faced with that I need somehow to read messages that was already received ...
But I cannot find the command that allow me to search in history or browse received messages.

Is there any possibility for this ?

**Comment by AsamK** [comment]
There's no message history, so messages that have already been received cannot be received again.

**Reply by redradist** [reply]
@AsamK But how I can see my messages in original client if nothing is saved ?

**Reply by AsamK** [reply]
The official clients store the message history locally on the device. signal-cli doesn't do that.

**Reply by redradist** [reply]
@AsamK Okay, gotcha, thanks !!
